### ISSUE TITLE
defineExtensionMessaging should be marked with `this: void`

### ISSUE THREAD

**Issue by minht11** [aklinker1/webext-core#82]
Basically the same idea as with https://typescript-eslint.io/rules/unbound-method/ so tooling knows destructoring is safe here.

https://github.com/aklinker1/webext-core/blob/ce0ad0d78de9feff2c91a239cd42294d989e6819/packages/messaging/src/extension.ts#L49-L51
```ts
function defineExtensionMessaging(this: void, condig: Config)
```

**Comment by minht11** [comment]
Currently that lint rule fails for me, hence this issue report.
I think I was bit wrong `this` should not be declared on `defineExtensionMessaging` but on `sendMessage` instead https://github.com/aklinker1/webext-core/blob/ce0ad0d78de9feff2c91a239cd42294d989e6819/packages/messaging/src/generic.ts#L40-L59

Not 100% sure but interface/class when used as a type behaves almost the same, so when it sees GenericMessenger interface which you try destructoring from it thinks `this` could belong to GenericMessenger and not be void

**Comment by aklinker1** [comment]
Yeah ok, that makes more sense. If you want to open a PR, that'd be great
